domain/application/service/application.go

```diff
@@ -775,6 +781,40 @@ func (s *Service) SetUnitPassword(ctx context.Context, unitName coreunit.Name, p
 	})
 }
 
+// GetUnitWorkloadStatus returns the workload status of the specified unit, returning an
+// error satisfying [applicationerrors.UnitNotFound] if the unit doesn't exist.
+func (s *Service) GetUnitWorkloadStatus(ctx context.Context, unitName coreunit.Name) (*corestatus.StatusInfo, error) {
+	if err := unitName.Validate(); err != nil {
+		return nil, errors.Trace(err)
+	}
+	unitUUID, err := s.st.GetUnitUUIDByName(ctx, unitName)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	agentStatus, err := s.st.GetUnitWorkloadStatus(ctx, unitUUID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return decodeUnitWorkloadStatus(agentStatus)
+}
+
+// SetUnitWorkloadStatus sets the workload status of the specified unit, returning an
+// error satisfying [applicationerrors.UnitNotFound] if the unit doesn't exist.
+func (s *Service) SetUnitWorkloadStatus(ctx context.Context, unitName coreunit.Name, status *corestatus.StatusInfo) error {
+	if err := unitName.Validate(); err != nil {
+		return errors.Trace(err)
+	}
+	workloadStatus, err := encodeUnitWorkloadStatus(status)
+	if err != nil {
+		return internalerrors.Errorf("encoding workload status: %w", err)
+	}
+	unitUUID, err := s.st.GetUnitUUIDByName(ctx, unitName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return s.st.SetUnitWorkloadStatus(ctx, unitUUID, workloadStatus)
```

Comment:
Same here.

Reply:
`SetUnitWorkloadStatus` makes sense to take a UUID, because the implementation uses the exiting private `setUnitWorkloadStatus`, which already has a couple of callers and takes a uuid.
https://github.com/juju/juju/blob/8111f59ad7af3eabacf4a0b49e251cb1fc187ced/domain/application/state/application.go#L1758-L1796

This makes `SetUnitWorkloadStatus` itself very simple, which I thought would trump a simpler service method.

And then for `GetUnitWorkloadStatus`, consistency here was my main priority.